backend/src/main/java/atlantbh/restaurants/controllers/RestaurantController.java

```diff
@@ -0,0 +1,46 @@
+package atlantbh.restaurants.controllers;
+
+import atlantbh.restaurants.models.Restaurant;
+import atlantbh.restaurants.models.filters.RestaurantFilterBuilder;
+import atlantbh.restaurants.models.sortkeys.RestaurantSortKeys;
+import atlantbh.restaurants.services.RestaurantService;
+import org.hibernate.service.spi.ServiceException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class RestaurantController extends BaseController<Restaurant, RestaurantService> {
+
+    public ResponseEntity filter(@RequestParam(value = "q", required = false) String query,
+                                 @RequestParam(value = "priceRange", required = false) Integer priceRange,
+                                 @RequestParam(value = "pageSize", required = true) Integer pageSize,
+                                 @RequestParam(value = "pageNumber", required = true) Integer pageNumber,
+                                 @RequestParam(value = "sortKey", required = false) String sortKey,
+                                 @RequestParam(value = "name", required = false) String name,
+                                 @RequestParam(value = "category", required = false) String category,
+                                 @RequestParam(value = "cousine", required = false) String cousine,
+                                 @RequestParam(value = "location", required = false) String location) {
+        try {
+            RestaurantFilterBuilder rfb = new RestaurantFilterBuilder()
+                    .setPriceRange(priceRange)
+                    .setQuery(query)
+                    .setName(name)
+                    .setCategory(category)
+                    .setCousine(cousine)
+                    .setLocation(location)
+                    .setPageSize(pageSize)
+                    .setPageNumber(pageNumber);
+            if (sortKey != null)
```

Comment:
you just need to call .setSortKey(getEnum(value, defaultValue)) .. but you can create method `protected <E extends Enum> E getEnum(String paramName, Class<E> enumType, E defaultValue);` in basecontroller.. So whenever you need to get some enum value from controller you can call that method.